third_party/move/move-vm/runtime/src/storage/layout_cache.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! Data structures for caching layouts of VM values ina long-living, concurrent cache. The design
+//! goal is to make sure that cache hit is semantically equivalent to layout construction. That is,
+//! if layout construction costs 5 units of gas, so does the cache hit. Failed layout construction
+//! never results in insertion into the cache.
+//!
+//! Additionally, caches store only roots of the layout tree. There is NO caching for sub-layouts
```

Comment:
You should be clearer what you mean with 'root', I saw earlier in this PR you only cache struct layouts, so it appears that is what you mean.

Reply:
Struct layouts and only top level, i.e. if you request layout of struct A with B inside, A is cached, but not B, so next access via B is a cache miss. Added a comment to make this clear